java-checks/src/main/java/org/sonar/java/checks/CollectionImplementationReferencedCheck.java

```diff
@@ -20,39 +20,45 @@
 package org.sonar.java.checks;
 
 import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
 import javax.annotation.Nullable;
 import org.sonar.check.Rule;
 import org.sonar.java.checks.helpers.QuickFixHelper;
-import org.sonarsource.analyzer.commons.collections.MapBuilder;
 import org.sonar.java.model.ModifiersUtils;
 import org.sonar.java.reporting.JavaQuickFix;
 import org.sonar.java.reporting.JavaTextEdit;
 import org.sonar.plugins.java.api.JavaFileScanner;
 import org.sonar.plugins.java.api.JavaFileScannerContext;
 import org.sonar.plugins.java.api.tree.BaseTreeVisitor;
+import org.sonar.plugins.java.api.tree.IdentifierTree;
+import org.sonar.plugins.java.api.tree.MemberSelectExpressionTree;
 import org.sonar.plugins.java.api.tree.MethodTree;
 import org.sonar.plugins.java.api.tree.Modifier;
 import org.sonar.plugins.java.api.tree.ModifiersTree;
 import org.sonar.plugins.java.api.tree.ParameterizedTypeTree;
 import org.sonar.plugins.java.api.tree.Tree;
 import org.sonar.plugins.java.api.tree.TypeTree;
 import org.sonar.plugins.java.api.tree.VariableTree;
+import org.sonarsource.analyzer.commons.collections.MapBuilder;
 
 @Rule(key = "S1319")
 public class CollectionImplementationReferencedCheck extends BaseTreeVisitor implements JavaFileScanner {
 
-  private static final String DEQUE = "java.util.Deque";
   private static final String LIST = "java.util.List";
-  private static final String MAP = "java.util.Map";
-  private static final String CONCURRENT_MAP = "java.util.concurrent.ConcurrentMap";
+  private static final String DEQUE = "java.util.Deque";
   private static final String QUEUE = "java.util.Queue";
   private static final String SET = "java.util.Set";
-  private static final String SORTED_MAP = "java.util.SortedMap";
   private static final String SORTED_SET = "java.util.SortedSet";
+  private static final String MAP = "java.util.Map";
+  private static final String CONCURRENT_MAP = "java.util.concurrent.ConcurrentMap";
+  private static final String SORTED_MAP = "java.util.SortedMap";
 
-  private static final Map<String, String> MAPPING = MapBuilder.<String, String> newMap()
+  private static final Map<String, String> SUGGESTED_INTERFACE_BY_COLLECTION_CLASS = MapBuilder.<String, String>newMap()
```

Comment:
We will probably need to rework some of this mapping strategy with the introduction of SequencedCollection, SequencedSet and and SequencedMap but let's keep this for when we can actually test against Java 21 samples